doc/over_react_redux_documentation.md

```diff
@@ -0,0 +1,320 @@
+1. Create your store and reducer. See the [example store](../web/over_react_redux/examples/simple/store.dart) for 
+ways to do this.
+1. Import OverReact Redux and your store into your `index.dart`.
+    ```dart
+    import 'package:over_react/over_react_redux.dart';
+    ```
+1. Wrap your component tree in a `ReduxProvider` and pass in the store.
+    ```dart
+    import 'package:over_react/over_react_redux.dart';
+    import 'package:over_react/react_dom.dart' as react_dom;
+
+    main() {
+      react_dom.render(
+        (ReduxProvider()..store = fooStore)(
+          // ... React component tree with connected components
+        ),
+        mountNode,
+      );
+    }
+    ```
+1. Import OverReact Redux and your store into the file with your component.
+1. Update your component class.
+    1. Add a new `UiFactory`, without the usual annotation, and wrap it with `connect()`.
+    1. Add the `ConnectPropsMixin` to your props class if needed. This mixin is needed if either of the following are true:
+        - You _do not_ use the `mapDispatchToProps` parameter on `connect`.
+        - You _do_ use `mapDispatchToProps` but want access to the `props.dipsatch` function. When using 
+        `mapDispatchToProps`, by default the `dispatch` property on `props` is removed. It can be added back 
+        using `ConnectPropsToMixin`. 
+    ```dart
+    // AppState is a class that represents the application's state and can be defined in the same file as the store.
+    UiFactory<FooProps> ConnectedFoo = connect<AppState, FooProps>()(Foo);
+    
+    @Factory()
+    UiFactory<FoodProps> Foo = _$Foo;
+    
+    // Use the ConnectPropsMixin to gain access to React Redux's dispatch function, which can be accessed via 
+    // props.dispatch.
+    @Props()
+    class _$FooProps extends UiProps with ConnectPropsMixin {
+        ...
+    }
+    
+    @Component2()
+    class FooComponent extends UiComponent2<FooProps> {
+        ...
+    }
+    ```
+    
+## ReduxProvider
+Redux uses React 16's `Context` API to pass information through the component tree. Consequently, the tree must be 
+wrapped in a `Provider` to handle the context.
+
+For this purpose, OverReact Redux has the `ReduxProvider` component. It is required to take in a store instance, and can
+ optionally take in a context object when using multiple stores.
+ 
+ __Example:__
+ ```dart
+react_dom.render(
+  (ReduxProvider()..store = fooStore)(
+    // ... React component tree with connected components
+  ),
+  mountNode,
+);
+```
+    
+## `connect`
+### Overview
+A wrapper around the JS react-redux `connect` function that supports OverReact components.
+
+> See: <https://react-redux.js.org/api/connect>
+
+__Example:__
+```dart
+UiFactory<CounterProps> ConnectedCounter = connect<CounterState, CounterProps>(
+    mapStateToProps: (state) => (
+      Counter()..count = state.count
+    ),
+    mapDispatchToProps: (dispatch) => (
+      Counter()..increment = () => dispatch(IncrementAction())
```

Comment:
#nit best practice is to wrap arrow functions in parens so they don't mess up the cascade
```suggestion
      Counter()..increment = (() => dispatch(IncrementAction()))
```